git/sparse_checkout.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+URL=$1
+
+if [ -z "$1" ]; then
+  echo "Usage [REMOTE-URL]"
+  exit 1
+fi
+
+REMOTE="$(echo "$URL" | sed -n 's|\(.*github.com\)/\([^/]*\)/\([^/]*\)/\(.*\)|\1/\2/\3.git|p')"
+MAIN_DIR="$(echo "$URL" | sed -n 's|\(.*github.com\)/\([^/]*\)/\([^/]*\)/\(.*\)|\3|p')"
+SUBDIR="$(echo "$URL" | sed -n 's|\(.*github.com\)/\([^/]*\)/\([^/]*\)/\(.*\)|\4|p')"
+
+test -d "$MAIN_DIR" || mkdir "$MAIN_DIR"
+cd "$MAIN_DIR" || exit
+test -d .git || git init --initial-branch=main
```

Comment:
Can get branch from param? If not, default to main.

Reply:
I've made this change, can I resolve this conversation?